netcore/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -54,18 +54,16 @@ partial class Environment
 
 		public static void FailFast (string message)
 		{
-			throw new NotImplementedException ();
+			FailFast (message, null, null);
 		}
 
 		public static void FailFast(string message, Exception exception)
 		{
-			throw new NotImplementedException ();
+			FailFast (message, exception, null);
 		}
 
-		public static void FailFast (string message, Exception exception, string errorMessage)
-		{
-			throw new NotImplementedException ();
-		}
+		[MethodImplAttribute (MethodImplOptions.InternalCall)]
+		public extern static void FailFast (string message, Exception exception, string errorSource);
```

Comment:
Does C# have default parameters? Can you say just one function:
```
public extern static void FailFast (string message, Exception exception = null, string errorSource = null);
```

Reply:
This is public API, the shape cannot be changed.